scripts/parse_tools/parse_checkers.py

```diff
@@ -354,13 +354,15 @@ def check_fortran_type(typestr, error=False):
         match = registered_fortran_ddt_name(typestr)
         dt = " derived"
     # End if
-    if match is None:
-        if error:
-            raise CCPPError("'{}' is not a valid{} Fortran type".format(typestr, dt))
-        else:
-            typestr = None
-        # End if
-    # End if
+    # DH* 20190913 - skip checking if a DDT is registered at this time
+    #if match is None:
+    #    if error:
+    #        raise CCPPError("'{}' is not a valid{} Fortran type".format(typestr, dt))
+    #    else:
+    #        typestr = None
+    #    # End if
+    ## End if
+    # *DH 20190913
```

Comment:
I'd suggest to worry about this once capgen is merged into master - until then, the logic in the master-version of the scripts that capgen would be using would require this change as well, because it assumes that (a) all DDTs used are defined in the same metadata file (which is certainly not going to be the case) and that they are defined in the metadata file before they are used. I understand that the feature/capgen version completes the analysis phase and then knows about all DDTs, but the current version in the master branch does not.

Reply:
I will accept that, however, this means that master and feature/capgen will diverge until then. Adding more logic to keep the if statement from being triggered in master, will allow updates to feature/capgen.